deeplearning4j-nn/src/main/java/org/deeplearning4j/datasets/iterator/EarlyTerminationDataSetIterator.java

```diff
@@ -0,0 +1,124 @@
+package org.deeplearning4j.datasets.iterator;
+
+import org.nd4j.linalg.dataset.DataSet;
+import org.nd4j.linalg.dataset.api.DataSetPreProcessor;
+import org.nd4j.linalg.dataset.api.iterator.DataSetIterator;
+
+import java.util.List;
+
+/**
+ * Builds an iterator that terminates once the number of minibatches returned with .next() is equal to a specified number
+ * Note that a call to .next(num) is counted as a call to return a minibatch regardless of the value of num
+ * This essentially restricts the data to this specified number of minibatches.
+ */
+public class EarlyTerminationDataSetIterator implements DataSetIterator {
+
+    private DataSetIterator underlyingIterator;
+    private int terminationPoint;
+    private int minibatchCount = 0;
+
+    /**
+     * Constructor takes the iterator to wrap and the number of minibatches after which the call to hasNext()
+     * will return false
+     * @param underlyingIterator, iterator to wrap
+     * @param terminationPoint, minibatches after which hasNext() will return false
+     */
+    public EarlyTerminationDataSetIterator(DataSetIterator underlyingIterator, int terminationPoint) {
+        if (terminationPoint <= 0 ) throw new IllegalArgumentException("Termination point (the number of calls to .next() or .next(num)) has to be > 0");
+        this.underlyingIterator = underlyingIterator;
+        this.terminationPoint = terminationPoint;
+    }
+
+    @Override
+    public DataSet next(int num) {
+        if (minibatchCount < terminationPoint) {
```

Comment:
if minibatchCount > (terminationPoint - num), before next(num) this call with succeed, returning a total # of elements strictly greater than terminationPoint